addons/resource/models/resource_calendar.py

```diff
@@ -508,8 +507,11 @@ def _leave_intervals_batch(self, start_dt, end_dt, resources=None, domain=None,
             resources_list = list(resources) + [self.env['resource.resource']]
         if domain is None:
             domain = [('time_type', '=', 'leave')]
-        if not any_calendar:
-            domain = domain + [('calendar_id', 'in', [False, self.id])]
+        if self:
+            domain = domain + [('calendar_id', 'in', [False] + self.ids)]
+        else:
+            domain = domain + [('calendar_id', '=', False)]
```

Comment:
Actually, if we do that we will alter the behavior which could be annoying in HR apps.

Maybe we could instead adding that additional condition in the domain given in parameter when we call that method in our code?

What do you think? 🙂

Reply:
yes sure, it's better